AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/ui/fragment/AppFragment.java

```diff
@@ -0,0 +1,73 @@
+package com.example.appactionvisualizer.ui.fragment;
+
+import android.content.Context;
+import android.os.Bundle;
+
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.example.appactionvisualizer.R;
+import com.example.appactionvisualizer.constants.Constant;
+import com.example.appactionvisualizer.databean.ActionType;
+import com.example.appactionvisualizer.ui.adapter.AppRecyclerViewAdapter;
+
+/**
+ * A fragment of MainActivity
+ * Displays apps with the same action type using recyclerview
+ */
+public class AppFragment extends Fragment {
+  private final static String TAG = "AppFragment";
+
+  private ActionType actionType = null;
+
+  /**
+   * Mandatory empty constructor for the fragment manager to instantiate the
+   * fragment (e.g. upon screen orientation changes).
+   */
+  public AppFragment() {
+  }
+
+  /**
+   * @param pos create a specific type fragment instance
+   * @return
+   */
+  @SuppressWarnings("unused")
+  public static AppFragment newInstance(int pos) {
+    AppFragment fragment = new AppFragment();
+    if(pos != 0) {
+      Bundle args = new Bundle();
+      args.putInt(Constant.ACTION_TYPE, pos - 1);
```

Comment:
instead of using putInt, what about putString(Constant.Action_TYPE, ActionType.getActionTypeValue(pos)).

we will implement getActionTypeValue(), it will return a String. 

Also, if you plan to use integer to map ActionType somehow, you probably want to make the constructor of your ActionType to be something like:

public ActionType(String value, int index) {
}

Reply:
Good catch. A new method is added to ActiontType class to act as a clearer implementation of mapping page index to ActionType.